### PR TITLE
fix(beta-labels): add beta labels

### PR DESCRIPTION
**Pull request by jpezninjo**
[SOCIALPLAT-434](https://mozilla-hub.atlassian.net/browse/SOCIALPLAT-434)

Add labels in 2 spots

- [x] Beta label in the footer on desktop, with link to feedback form
- [x] Beta label in the header on mobile
- [x] Light and dark mode (the default accent color supports light/dark mode, and the text in these labels changes too)

Footer (light mode)
image

Header (dark home)
image

Header (dark conversations)
image

Example of undesired behavior - beta label shows up in settings sub-page
image

Corrected UI
image

### CODE REVIEW

**Comment by jpezninjo** [comment]
I did so much figuring out for this amount of code :smiling_face_with_tear:

File: pages/settings/about/index.vue [hunk 1]
```diff
@@ -17,7 +17,7 @@ function handleShowCommit() {
 </script>
 
 <template>
-  <MainContent back-on-small-screen>
+  <MainContent back-on-small-screen no-beta-label>
```
Comment (wtfluckey) [inline pages/settings/about/index.vue:20]: Another Vue nuance!! Idk why the [recommendation](https://v2.vuejs.org/v2/style-guide/#Prop-name-casing-strongly-recommended) to declare props in camelCase, but then use them in kebab-case in the HTML bugs me SOOOOOO much, but it does.

File: components/main/MainContent.vue [hunk 2]
```diff
@@ -63,3 +68,20 @@ const containerClass = computed(() => {
...
 </template>
+
+<style>
+  .beta {
+    width: 56.16px;
+    height: 35.16px;
+    padding: 5px 6px;
+    border-radius: 8px;
+    border: 2px solid var(--c-primary);
+    gap: 17.70120620727539px;
```
Comment (wtfluckey) [inline components/main/MainContent.vue:79]: Where did this value come from? Does it need to be so specific?

File: components/nav/NavFooter.vue [hunk 3]
```diff
@@ -38,3 +43,23 @@ function toggleDark() {
...
+    padding: 8px;
+    gap: 8px;
+    line-height: 17.8px;
+
+    color: var(--c-text-base-light);
+    font-size: 13.13px;
+    /* font-weight: 700; */
+    font-weight: 510;
+    a {
+    }
```
Comment (wtfluckey) [inline components/nav/NavFooter.vue:63]: Got a few things up here. Need to remove the comment and the empty `a`, and `510` is an invalid value for `font-weight`. I think we want 500 here.

File: components/main/MainContent.vue (hunk 2 above)
Comment (jpezninjo) [reply components/main/MainContent.vue:79]: Oooh I took this giant glob from Figma. I'm realizing we don't even need `gap`. Will remove

File: components/nav/NavFooter.vue (hunk 3 above)
Comment (jpezninjo) [reply components/nav/NavFooter.vue:63]: *facepalm* I missed this when I made some other change AND when I did my self-review. Good catch! I'll remove the comment and rule.

I'll go ahead and make that 500. My Firefox inspector wasn't complaining and it does a change when I play around with the 510 value.

File: components/main/MainContent.vue (hunk 2 above)
Comment (jpezninjo) [reply components/main/MainContent.vue:79]: I'm still laughing at how ridiculous this must be to come across in code

**Comment by wtfluckey** [comment]
Looks like we might have the background colors set incorrectly for the themes. The text color contrast is failing accessibility in the footer beta label on light mode

Screenshot 2023-08-16 at 11 16 24 AM

File: components/main/MainContent.vue (hunk 2 above)
Comment (wtfluckey) [reply components/main/MainContent.vue]: Ahh yeah, I thought these looked like Figma styles. Generally, I would recommend not relying on CSS that is generated from tools like Figma. Maybe the generators are better today than they used to be and I'm just being old, but they are usually unreliable, operate in isolation, don't take into account our patterns or best practices and encourage bad habits.

I put together #27 to illustrate how many styles Figma generated that we actually don't need. I'm happy to walk through any of this with you too!

File: components/main/MainContent.vue [hunk 4]
```diff
@@ -63,3 +68,19 @@ const containerClass = computed(() => {
     </div>
   </div>
 </template>
+
+<style>
+  .beta {
+    width: 56.16px;
+    height: 35.16px;
```
Comment (collectedmind) [inline components/main/MainContent.vue:75]: Why the partial pixels?

File: components/main/MainContent.vue [hunk 5]
```diff
@@ -63,3 +68,19 @@ const containerClass = computed(() => {
...
+    border-radius: 8px;
+    border: 2px solid var(--c-primary);
+
+    color: var(--c-primary);
+    font-size: 18px;
+    font-weight: 700;
+    line-height: 21px;
+    letter-spacing: 0em;
+  }
+</style>
```
Comment (collectedmind) [inline components/main/MainContent.vue:86]: Also, Is there a reason we are switching to styles like this as opposed to sticking with the unoCSS paradigm? `border-r-8` as opposed to a specific.  Don't get me wrong, I am all for styles over framework, but I would love to better understand if there is a need to mix it here?

**Comment by jpezninjo** [comment]
@wtfluckey I only added the text color changing. In my screenshots, I've set my theme color to the lighter purple. Did you set your theme color yet?

@aaga is there a way to revert to 'unset'?

This is what I'd expect the user to see if they haven't set their theme -


Light
image



Dark
image

File: components/nav/NavFooter.vue [hunk 6]
```diff
@@ -9,6 +9,11 @@ function toggleDark() {
 
 <template>
   <footer p4 text-sm text-secondary-light flex="~ col">
+    <div mb-15px class="beta-label">
+      <b>Mozilla Social is an experimental beta product.</b> We would love <NuxtLink href="https://survey.alchemer.com/s3/7470063/mozilla-social-feedback-form" target="_blank" external>
+        <u>your feedback</u>
+      </NuxtLink>.
+    </div>
```
Comment (collectedmind) [inline components/nav/NavFooter.vue:16]: We should sort out how to get this ready for localization.  Having the NuxtLink in there might make this ... interesting 😉

File: components/nav/NavFooter.vue [hunk 7]
```diff
@@ -38,3 +43,20 @@ function toggleDark() {
...
+    border-radius: 8px;
+    padding: 8px;
+    gap: 8px;
+    line-height: 17.8px;
+
+    color: var(--c-text-base-light);
+    font-size: 13.13px;
+    font-weight: 500;
+  }
+</style>
```
Comment (collectedmind) [inline components/nav/NavFooter.vue:62]: Similar question re: UnoCss.  Just want to be wary of mixing patterns, even if another pattern is more comfortable to work in

**Review by collectedmind — COMMENTED** [review]
Couple of questions and one extra fun localization challenge.

File: components/main/MainContent.vue (hunk 4 above)
Comment (jpezninjo) [reply components/main/MainContent.vue:75]: I took these from Figma... and I'm assuming I'm supposed to round these. I'll make that change and will do so moving forward. I was trying to do the least amount of work and thought you guys would raise it in PR comments if it should be rounded :grimacing:

File: components/main/MainContent.vue (hunk 5 above)
Comment (jpezninjo) [reply components/main/MainContent.vue:86]: I had not realized there was a border rule. I have not dived too deeply into the rules beyond positioning rules. I'll go through rn and reply back to this thread

File: components/nav/NavFooter.vue (hunk 6 above)
Comment (aaga) [reply components/nav/NavFooter.vue:16]: Check out the `i18n-t` component; it's designed for situations like this: https://vue-i18n.intlify.dev/guide/advanced/component.html

I used it in my Report Modal PR, so you can look at that again as an example.

As Joel predicted, it is a little... interesting.

File: components/main/MainContent.vue (hunk 5 above)
Comment (jpezninjo) [reply components/main/MainContent.vue]: I was only able to find `border-rounded`, which produces a less fun shape - 
image


I'm preferenced to keep our own border-rule. I think it makes a big difference not here but on the Footer's label

File: components/nav/NavFooter.vue (hunk 6 above)
Comment (jpezninjo) [reply components/nav/NavFooter.vue]: @aaga was able to figure it out, thanks!! That wasn't _too_ bad, we did similar in the Pocket repo

File: components/nav/NavFooter.vue (hunk 7 above)
Comment (jpezninjo) [reply components/nav/NavFooter.vue]: Aly helped to remove lots of these rules, and I removed a couple more

File: components/main/MainContent.vue (hunk 5 above)
Comment (wtfluckey) [reply components/main/MainContent.vue]: I was able to figure this out and will have a PR soon!

File: components/nav/NavFooter.vue [hunk 8]
```diff
@@ -9,6 +11,14 @@ function toggleDark() {
 
 <template>
   <footer p4 text-sm text-secondary-light flex="~ col">
+    <div mb-15px p10px font-500 class="beta-label">
+      <i18n-t keypath="footer.beta.description">
+        <b>{{ $t('footer.beta.desc_1') }}</b>
+        <NuxtLink href="https://survey.alchemer.com/s3/7470063/mozilla-social-feedback-form" target="_blank" external>
+          <u>{{ $t('footer.beta.link') }}</u>
+        </NuxtLink>.
```
Comment (wtfluckey) [inline components/nav/NavFooter.vue:19]: Do we end up with double periods since we have one on the description value in `en.json` (line 177)?

Comment (jpezninjo) [reply components/nav/NavFooter.vue]: REALLY GOOD CATCH! This is the thing I was talking about over Zoom!  This won't be used unless we add in a `{2}` to make `{0} We would love {1}.{2}` (I played around with it)

I will remove the stray period here, though.

**Review by wtfluckey — APPROVED** [review]
Excellent work here!

File: components/nav/NavFooter.vue (hunk 8 above)
Comment (wtfluckey) [reply components/nav/NavFooter.vue]: AHA! It treats it as another child...lol!

**Review by aaga — APPROVED** [review]
This looks mighty fine!

**Pull request merged**